### ISSUE TITLE
[Feature/Enhancement] Generate Shorter Image Names

### ISSUE THREAD

**Issue by stexyz** [dvorka/mindforger#1417]
**When pasting images don't append :**

When having multiple images pasted the filenames get really long: 
![image](Tasks.__________image.png)
![image](Tasks.___________image.png)

Additionally, when pasting the image for several times (trying to create a better shot etc.) and deleting the image references the files are kept. So typically I try couple of takes and end up with image name ______image and 5 image files in my repo that are not referenced.
...

Possible solution:

1. Naming - append numbers not underscores.
2. When saving changes check for files not referenced and remove them. Perhaps also fix the naming to always keep a full sequence of numbers.

**Comment by Iznik285** [comment]
I agree with the above. There are some further thoughts here: https://github.com/dvorka/mindforger/issues/1303

**Comment by stexyz** [comment]
An additional thought - consider including the note name into the image name to increase readability - currently it only uses the notebook name.

**Comment by dvorka** [comment]
Fix hints - use `<O name>.<N name>.<image name>_<number>.png`:

* use existing method which instead of duplication of `_` adds number to make it unique
* include N name (possibly configurable)

However, extra `_` might be cause by filesystem adversarial characters which are escaped this way.
